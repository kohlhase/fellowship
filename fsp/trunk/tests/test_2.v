(* Coq script *) 
Parameter A B C : Prop. 
Theorem t1 : ((A \/ B) -> ((B -> C) -> (A \/ C))). 
Proof (fun (ab:(A \/ B)) => (fun (bc:(B -> C)) => (match ab with or_introl x1 => (let a:A := x1 in (or_introl _ a)) | or_intror x1 => (let b:B := x1 in (let c:C := (bc b) in (or_intror _ c))) end))). 
