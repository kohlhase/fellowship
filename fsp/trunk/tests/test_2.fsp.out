
    > Current logic: intuitionistic sequent calculus.
      
fsp < 
    > Current logic: intuitionistic sequent calculus.
      
fsp < 
    > Current logic: minimal intuitionistic sequent calculus.
      
fsp < fsp < 
    > A,B,C defined.
      
fsp < fsp < 
    > Proof term: 
      μthesis:A∨B⇒(B⇒C)⇒A∨C.<?1||thesis>
      Natural language:
      we need to prove A∨B⇒(B⇒C)⇒A∨C
        .... (1) <======
      done
      
      1 goal yet to prove!
      
      
       |-----  1
      *:A∨B⇒(B⇒C)⇒A∨C
      
fsp < 
    > Proof term: 
      μthesis:A∨B⇒(B⇒C)⇒A∨C.<λab:A∨B.λbc:B⇒C.?1.1.1||thesis>
      Natural language:
      we need to prove A∨B⇒(B⇒C)⇒A∨C
        assume A∨B (ab)
        assume B⇒C (bc)
        .... (1.1.1) <======
      done
      
      1 goal yet to prove!
      
      
      bc:B⇒C
      ab:A∨B
       |-----  1.1.1
      *:A∨C
      
fsp < 
    > 2 subgoals generated. 
      
      Proof term: 
      μthesis:A∨B⇒(B⇒C)⇒A∨C.<λab:A∨B.λbc:B⇒C.μac:A∨C.<ab||(?1.1.1.2.1,?1.1.1.2.2)>||thesis>
      Natural language:
      we need to prove A∨B⇒(B⇒C)⇒A∨C
        assume A∨B (ab)
        assume B⇒C (bc)
        we need to prove A∨C
          by ab
          by cases: 
            first case:
              by case hypothesis
            ...(1.1.1.2.1) <======
            second case:
              by case hypothesis
            ...(1.1.1.2.2)
      done
      
      2 goals yet to prove!
      
      
      *:A
      bc:B⇒C
      ab:A∨B
       |-----  1.1.1.2.1
      ac:A∨C
      
fsp < 
    > Proof term: 
      μthesis:A∨B⇒(B⇒C)⇒A∨C.<λab:A∨B.λbc:B⇒C.μac:A∨C.<ab||(?1.1.1.2.1,?1.1.1.2.2)>||thesis>
      Natural language:
      we need to prove A∨B⇒(B⇒C)⇒A∨C
        assume A∨B (ab)
        assume B⇒C (bc)
        we need to prove A∨C
          by ab
          by cases: 
            first case:
              by case hypothesis
            ...(1.1.1.2.1)
            second case:
              by case hypothesis
            ...(1.1.1.2.2) <======
      done
      
      2 goals yet to prove!
      
      
      *:B
      bc:B⇒C
      ab:A∨B
       |-----  1.1.1.2.2
      ac:A∨C
      
fsp < 
    > Closed a branch. 
      
      Proof term: 
      μthesis:A∨B⇒(B⇒C)⇒A∨C.<λab:A∨B.λbc:B⇒C.μac:A∨C.<ab||(?1.1.1.2.1,μ'b:B.<bc||b*μ'c:C.<R(c)||ac>>)>||thesis>
      Natural language:
      we need to prove A∨B⇒(B⇒C)⇒A∨C
        assume A∨B (ab)
        assume B⇒C (bc)
        we need to prove A∨C
          by ab
          by cases: 
            first case:
              by case hypothesis
            ...(1.1.1.2.1) <======
            second case:
              by case hypothesis
              we proved B (b)
              by bc
              and by b
              we proved C (c)
              by c
              trivial
            done
      done
      
      1 goal yet to prove!
      
      
      *:A
      bc:B⇒C
      ab:A∨B
       |-----  1.1.1.2.1
      ac:A∨C
      
fsp < 
    > Closed the last branch: 
      Proof completed!
      
      Proof term: 
      μthesis:A∨B⇒(B⇒C)⇒A∨C.<λab:A∨B.λbc:B⇒C.μac:A∨C.<ab||(μ'a:A.<L(a)||ac>,μ'b:B.<bc||b*μ'c:C.<R(c)||ac>>)>||thesis>
      Natural language:
      we need to prove A∨B⇒(B⇒C)⇒A∨C
        assume A∨B (ab)
        assume B⇒C (bc)
        we need to prove A∨C
          by ab
          by cases: 
            first case:
              by case hypothesis
              we proved A (a)
              by a
              trivial
            done
            second case:
              by case hypothesis
              we proved B (b)
              by bc
              and by b
              we proved C (c)
              by c
              trivial
            done
      done
      
      
fsp < 
    > t1 defined.
      
fsp < 
    > Firing up Coq for confirmation.
      
fsp < 
    > Firing up Coq for confirmation.
      
fsp < 
    > File ./test_2.nl written. Enjoy the reading.
      
fsp < 
    > EOF!

